### DISCUSSION TITLE
Incorporate VS Code ecosystem

### DISCUSSION THREAD

**Discussion by mdabidullah** [lapce/lapce#2688] [Ideas]
Hi,
First of all great job, I really like this editor and how fast it is. 

I was thinking about the work it would take to bring it to parity with VS Code. The main advantage VS Code has is it's ecosystem of extensions. Perhaps a shortcut might be to incorporate the VS Code ecosystem in Lapce. So any extension which works in VS Code can be automatically added to Lapce. This would be done by incorporating a javascript runtime (maybe Bun). Later once the editor gets more users, people will be invested to build native extensions. 

Advantages:
+ More users
+ Less friction
+ Huge ecosystem almost for free
+ Less power to big tech

Disadvantages:
- Performance hit (I don't know much about it, but may be Bun runtime could make it reasonably fast?)
- Maintenance

Anyway, just wanted to put the idea out there, thanks for reading.

**Comment by mdabidullah** [comment]
I just found out the extension system is built on WASI, so I guess just supporting the configuration of VS Code extensions would be enough?

**Comment by panekj** [comment]
We're not going to implement VSCode extensions as that would require us to depend on JS runtime

**Reply by mdabidullah** [reply]
I did suspect that would be a deal-breaker, thanks for the reply. Have a nice day :)
